pkg/spoke/submarineragent/config_controller.go

```diff
@@ -181,6 +191,49 @@ func (c *submarinerConfigController) sync(ctx context.Context, syncCtx factory.S
 		return c.updateGatewayStatus(syncCtx.Recorder(), config)
 	}
 
+	if config.Status.ManagedClusterInfo.Platform == "GCP" {
+		if IsSubmarinerEnvPrepared(c.configClient, config.Namespace, config.Name) {
```

Comment:
`IsSubmarinerEnvPrepared` can be simplified to:

```suggestion
		if meta.IsStatusConditionTrue(config.Status.Conditions, configv1alpha1.SubmarinerConfigConditionEnvPrepared) {
```